### DISCUSSION TITLE
Question about input files for QC of scATAC-seq data

### DISCUSSION THREAD

**Discussion by LilyYe98** [stuart-lab/signac#1060] [Q&A] [answered]
Hi,
Thanks for providing such helpful tool!

I want to do QC for the published data in https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM5155451.
But what they provide are barcodes.tsv.gz, matrix.mtx.gz, peaks.bed.gz and fragments.tsv.gz.
I wonder how to use them to do QC with signac. 
Because in signac tutorial, input files are matrix.h5 for counts, atac_v1_pbmc_10k_singlecell.csv for metadata. 
I do not understand how to use barcodes.tsv.gz, matrix.mtx.gz, peaks.bed.gz files to create variable counts and variable metadata.

I found some similar questions in Discussions like #481, you mentioned function `Seurat::Read10X()` or `Matrix::readMM()` if using matrix.mtx instead of .h5 file as input to Signac analysis. But I still do not understand how to use one of these 2 function to create variable counts and then pass this variable counts to `CreateChromatinAssay` function. Also in #120, you mentioned the last section in the [PBMC vignette](https://satijalab.org/signac/articles/pbmc_vignette.html) when we don't have .csv file from 10X genomics for metadata, but I still do not understand which file I should use to generate variable metadata and then pass it to  CreateSeuratObject function based on Signac tutorial.

It is my first time to use Signac and Seurat and I am sorry to ask such stupid question.

Looking forward to your reply! Thanks!

**Comment by timoast** [comment, marked as answer]
> I found some similar questions in Discussions like https://github.com/timoast/signac/issues/481, you mentioned function Seurat::Read10X() or Matrix::readMM() if using matrix.mtx instead of .h5 file as input to Signac analysis. But I still do not understand how to use one of these 2 function to create variable counts and then pass this variable counts to CreateChromatinAssay function.

The `Seurat::Read_10X()` function will return a count matrix, just like `Seurat::Read10X_h5()`. Once you have the counts matrix you can proceed with steps shown in the Signac vignettes to create the Seurat object and chromatin assay

> Also in https://github.com/timoast/signac/issues/120, you mentioned the last section in the [PBMC vignette](https://satijalab.org/signac/articles/pbmc_vignette.html) when we don't have .csv file from 10X genomics for metadata, but I still do not understand which file I should use to generate variable metadata and then pass it to CreateSeuratObject function based on Signac tutorial.

You can find this information in the documentation for the individual functions. You can add metadata to the Seurat object using the `AddMetaData()` function.
